tasks/master-setup.yml

```diff
@@ -45,6 +45,19 @@
   changed_when: "'created' in calico_result.stdout"
   when: kubernetes_pod_network.cni == 'calico'
 
+- name: "Get K8s version"
+  shell: kubectl version | base64 | tr -d '\n'
+  register: k8_version
+  when: kubernetes_pod_network.cni == 'weave'
+
+- name: Configure Weave networking.
+  command: "{{ item }}"
+  with_items:
+    - "kubectl apply -f https://cloud.weave.works/k8s/net?k8s-version={{ k8_version.stdout_lines[0] }}"
```

Comment:
This seems an odd way to do it, but that's the official instructions, so who am I to argue: https://www.weave.works/docs/net/latest/kubernetes/kube-addon/#-installation